chingu-voyage4/Bears-Team-1
# Added basic authentication

## Added Authentication
**This PR adds 2 different authentication strategies.**

- Google
- Github
- Facebook will be added until my account gets unlocked.

## Routes
**All auth routes have a parent route called `/auth`.**

- Google Route: `/auth/google`
- Github Route: `/auth/github`
- Future Facebook Route: `/auth/facebook`

## Mocha Tests
**Right now, there are 2 different tests that I added**

- Test to see if `/auth/google` returns a 302 code for a successful redirect.
- Test to see if `/auth/github` returns a 302 code for a successful redirect.
- Future test to see if `/auth/facebook` returns a 302 code for a successful redirect.

## File Structure
**I've added 2 new folders in the server folder for authentication**

- The `service` folder contains `passportAuth.js` which will redirect users to their chosen authentication method and provides the callback url to recieve the user data.

- The `authRoute` folder contains `authRoutes.js` which will handle all `/auth/` requests and initiates passport to begin authentication on the chosen auth strategy.

## How to test
- Run the command `npm run start-server`
- Once localhost is up and running, type `localhost:3001/auth/google` or `localhost:3001/auth/github` to test the authentication.
- Get the proper client IDs and client secrets from our trello board and then copy and paste the IDs and client secrets into your .env file.
- When you complete the authentication provided by either Google or Github, check your command console and you should see all of your user data on your console.
